prometheus/src/main/scala/filodb/prometheus/parse/Parser.scala

```diff
@@ -7,16 +7,12 @@ import filodb.query._
 
 trait BaseParser extends Expressions with JavaTokenParsers with RegexParsers with PackratParsers {
 
-  lazy val identifier: PackratParser[Identifier] = {
+  lazy val labelNameIdentifier: PackratParser[Identifier] = {
     "[a-zA-Z_][a-zA-Z0-9_]*".r ^^ { str => Identifier(str) }
   }
 
-  lazy val labelIdentifier: PackratParser[Identifier] = {
-    "[a-zA-Z_][a-zA-Z0-9_:\\-\\.]*".r ^^ { str => Identifier(str) }
-  }
-
-  lazy val string: PackratParser[Identifier] = {
-    "[a-zA-Z_][a-zA-Z0-9_:|`~!@$#%^&*()s+=?><:;{}-]*".r ^^ { str => Identifier(str) }
+  lazy val metricNameIdentifier: PackratParser[Identifier] = {
+    "[a-zA-Z_:][a-zA-Z0-9_:\\-\\.]*".r ^^ { str => Identifier(str) }
   }
 
   protected lazy val quotedSeries: PackratParser[Identifier] =
```

Comment:
Please rename to labelValue

Reply:
but it's not the label value.  It actually is the metric name expression.